lib/cli/src/frameworks/web-components/ts/Button.ts

```diff
@@ -0,0 +1,33 @@
+import { html } from 'lit-html';
+
+export interface ButtonProps {
```

Comment:
I wonder if we should make all properties here optional or `ButtonProps` should be exported as a partial type (`type ButtonProps = Partial<{...}>`). To simplify all the references to just ButtonProps

Reply:
Partial means you can skip mandatory properties. Which is not what we want

Reply:
It does but my point is everywhere we reference this, we use Partial anyway. So we should probably solve the problem at the interface here rather than every time it gets used (which could mean using partial here or just making the properties optional).

Reply:
We have the same interface for all frameworks. Do you have an example of where Partial would bring value ?

Reply:
I'm not arguing FOR partial im telling you its used already in this PR:

```ts
const Template: Story<Partial<ButtonProps>> = (args) => Button(args);
```

in the stories. 

which in every other framework's stories would instead be:

```ts
const Template: Story<ButtonProps> = (args) => Button(args);
```

so i was just trying to point out that maybe instead of using partial, we should make the optional properties in `ButtonProps` optional. If that's all of them, fine, or consider using partial when defining the type rather than when using it.